app/views/videos/show.html.haml

```diff
@@ -17,7 +17,14 @@
         %hr
         .actions
           %a.btn.btn-primary(href="") Watch Now
-          =link_to "+My Queue", user_queue_items_path(current_user, video_id: @video.id), method: :post, class: "btn btn-success"
+          -if !current_user.queue_items.pluck(:video_id).include?(@video.id)
```

Comment:
Actually, this logic should be moved to a helper method where you pass in the user and video.